index.js

```diff
@@ -81,7 +83,14 @@ module.exports = function (opts) {
 					deleteRequireCache(modId);
 				});
 			}
-			jasmine.addReporter(new SilentReporter(cb, errorOnFail));
+			jasmine.addReporter(new SilentReporter(function (error) {
+				if (error) {
+					cb(error);
+				} else {
+					self.emit('end');
+					cb();
```

Comment:
@phated Thanks for taking the time.

Ok, I've been reading to try to better understand the issue here, but I'm still confused. My understanding is that we're creating a transform stream, through the use of through2 and readable-stream, which is supposed to be a re-implementation of streams3. From the documentation, a transform stream is supposed to be both a readable AND a writable stream. So my understanding is that we already return a writable stream.

After reading your comment and re-reading the docs for the nth time, I thought that maybe the problem was that the data never starts flowing because we're missing the "data" handler or because we don't pipe to a writable stream ([ref](https://nodejs.org/dist/latest-v6.x/docs/api/stream.html#stream_class_stream_readable)), but if that was the case the flush event would never be called, and it is since that's where I emit the end event.

I'm so sorry if I'm a little dense. I really, really want to understand streams.